components/post_view/post_list/post_list.jsx

```diff
@@ -8,12 +8,35 @@ import LoadingScreen from 'components/loading_screen';
 import {PostRequestTypes} from 'utils/constants';
 
 import {getOldestPostId, getLatestPostId} from 'utils/post_utils';
+import {clearMarks, mark, measure, trackEvent} from 'actions/telemetry_actions.jsx';
 
 import VirtPostList from 'components/post_view/post_list_virtualized/post_list_virtualized';
 
 const MAX_NUMBER_OF_AUTO_RETRIES = 3;
 export const MAX_EXTRA_PAGES_LOADED = 10;
 
+// Measures the time between channel or team switch started and the post list component rendering posts.
+// Set "fresh" to true when the posts have not been loaded before.
+function markAndMeasureChannelSwitchEnd(fresh = false) {
```

Comment:
Note that fresh may not always be exactly what we expect here since posts could have been preloaded in the background if we're switching to an unread channel

Reply:
Got it. Good to consider when analyzing the data.